xcube_cci/chunkstore.py

```diff
@@ -474,56 +474,35 @@ def _adjust_chunk_sizes(cls, chunks, sizes, time_dimension):
         if cls._is_of_acceptable_chunk_size(sum_chunks):
             chunks_are_acceptable = True
             for i in range(len(chunks)):
-                if i != time_dimension and chunks[i] > sizes[i]:
+                if i != time_dimension and (chunks[i] > sizes[i] or sizes[i] % chunks[i] > 0):
                     chunks_are_acceptable = False
                     break
             if chunks_are_acceptable:
                 return chunks
         # determine valid values for a chunk size.
         # A value is valid if the size can be divided by it without remainder
         valid_chunk_sizes = []
-        best_chunks = chunks.copy()
+        # best_chunks = chunks.copy()
         for i, chunk, size in zip(range(len(chunks)), chunks, sizes):
             # do not rechunk time dimension
             if i == time_dimension:
                 valid_chunk_sizes.append([chunk])
                 continue
-            valid_dim_chunk_sizes = []
-            if sum_chunks > _MAX_CHUNK_SIZE:
-                valid_dim_chunk_sizes = common_divisors(chunk)
-            else:  # sum_chunks < _MIN_CHUNK_SIZE
-                # handle case that the size cannot be
-                # divided evenly by the chunk
-                if size % chunk > 0:
-                    if np.prod(chunks, dtype=np.int64) / chunk * size \
-                            < _MAX_CHUNK_SIZE:
-                        # if the size is small enough to be ingested
-                        # in single chunk, take it
-                        valid_dim_chunk_sizes.append(size)
-                    else:
-                        # otherwise, give in to that we cannot
-                        # chunk the data evenly
-                        valid_dim_chunk_sizes += \
-                            (list(range(chunk, size + 1, chunk)))
-                    valid_chunk_sizes.append(valid_dim_chunk_sizes)
-                    continue
-                for r in range(chunk, size + 1, chunk):
-                    if size % r == 0:
-                        valid_dim_chunk_sizes.append(r)
+            valid_dim_chunk_sizes = common_divisors(size)
             valid_chunk_sizes.append(valid_dim_chunk_sizes)
         # recursively determine the chunking most similar to the original
         # file chunking with an acceptable size
-        orig_indexes = [0] * len(best_chunks)
         initial_best_chunks = [vcs[-1] for vcs in valid_chunk_sizes]
-        best_deviation = math.inf
+        initial_test_chunks = [vcs[0] for vcs in valid_chunk_sizes]
+        best_deviation = 0
         chunks, chunk_size, chunk_deviation = cls._get_best_chunks(
-            chunks, valid_chunk_sizes, initial_best_chunks, orig_indexes, 0, 0,
-            time_dimension, best_deviation
+            initial_test_chunks, valid_chunk_sizes, initial_best_chunks,
```

Comment:
why is the param called initial_test_chunks? The module where it is located is not a testing one.

Reply:
Because it is tested how good these chunks are.